services/src/main/java/org/keycloak/services/resources/LoginActionsService.java

```diff
@@ -868,13 +868,13 @@ public Response processConsent(final MultivaluedMap<String, String> formData) {
 
         for (String clientScopeId : authSession.getClientScopes()) {
             ClientScopeModel clientScope = KeycloakModelUtils.findClientScopeById(realm, client, clientScopeId);
-            if (clientScope != null && clientScope.isDisplayOnConsentScreen()) {
-                if (!grantedConsent.isClientScopeGranted(clientScope)) {
+            if (clientScope != null) {
+                if (!grantedConsent.isClientScopeGranted(clientScope) && clientScope.isDisplayOnConsentScreen()) {
                     grantedConsent.addGrantedClientScope(clientScope);
                     updateConsentRequired = true;
                 }
             } else {
-                logger.warnf("Client scope with ID '%s' not found", clientScopeId);
```

Comment:
I changed this line because _authSession.getClientScopes()_ does not include only client scopes but also clients.